stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -243,8 +231,10 @@ public void onHeaders(Metadata.Headers headers) {
     public void onPayload(RespT payload) {
       observer.onValue(payload);
 
-      // Request delivery of the next inbound message.
-      call.request(1);
+      if (streamingResponse) {
```

Comment:
Throwing an exception will cancel the call and should eventually log the exception. I think that's the only way you can cancel the call from here.